purchase_freight_delay/models/purchase.py

```diff
@@ -0,0 +1,69 @@
+# © 2022 David BEAL @ Akretion
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from datetime import timedelta
+from odoo import api, fields, models
+from odoo.tools import DEFAULT_SERVER_DATETIME_FORMAT
+
+
+class PurchaseOrder(models.Model):
+    _inherit = "purchase.order"
+
+    freight_rule_id = fields.Many2one(comodel_name="freight.rule", copy=False)
+    freight_duration = fields.Integer(
+        compute="_compute_freight_duration",
+        inverse="_inverse_freight_duration",
+        store=True,
+        copy=False,
+    )
```

Comment:
It's missing a field like : 
```    date_dispatch = fields.Datetime('Date of dispatch', help="When the goods exit the boarding place" )
```

freight_duration:
* default from freight_rule
* can be set manually

date_dispatch
*  default from date_planned - freight_duration
*  can be set manually

date_planned:
* if date_dispatch or freigth_duration are changed then date_planned = freigth_duration + date_dispatch

Reply:
Our process:
 
- (we) the client, agree with the supplier on a date of dispatch. It's part of the contract, and should be clearly displayed on the PO. The supplier is engaged to deliver the goods at this adress, on this date.
- we then choose the freight method. (and get a estimative freigth_duration).
- the freigth duration may be changed manually
- the date_planned is computed based on duration + date of shipment

There is also "normal" cases: the buyer set only the date_planned.

So for the buyer, it's quite clear, and he expects two fields:
- date planned = date when the goods should be in the destination warehouse
- date dispatch = date when the goods should be in the source warehouse